src/net/sourceforge/kolmafia/swingui/panel/GearChangePanel.java

```diff
@@ -0,0 +1,1496 @@
+    if (type != ItemDatabase.getConsumptionType(item.getItemId())) {
+      return false;
+    }
+
+    // Make sure we meet requirements in Limitmode, otherwise show (greyed out)
+    return KoLCharacter.getLimitMode() == LimitMode.NONE
+        || EquipmentManager.canEquip(item.getName());
+  }
+
+  private AdventureResult currentOrSelectedItem(final int slot) {
+    AdventureResult item = (AdventureResult) this.equipment[slot].getSelectedItem();
+    return (item != null) ? item : EquipmentManager.getEquipment(slot);
+  }
+
+  private static void updateEquipmentList(
+      final LockableListModel<AdventureResult> currentItems,
+      final List<AdventureResult> newItems,
+      final AdventureResult equippedItem) {
+    currentItems.retainAll(newItems);
+    newItems.removeAll(currentItems);
+    currentItems.addAll(newItems);
+    currentItems.setSelectedItem(equippedItem);
+  }
+
+  private void updateEquipmentModelsInternal(final List<List<AdventureResult>> equipmentLists) {
+    // For all the slots that we maintain a custom list, update the model specially
+    for (int slot = 0; slot < EquipmentManager.ALL_SLOTS; ++slot) {
+      LockableListModel<AdventureResult> model = equipmentModels.get(slot);
+      if (model == null) {
+        continue;
+      }
+
+      if (slot == EquipmentManager.WEAPON || slot == EquipmentManager.OFFHAND) {
+        if (KoLCharacter.inFistcore()) {
+          this.equipment[slot].setEnabled(false);
+          continue;
+        }
+      }
+
+      List<AdventureResult> items = equipmentLists.get(slot);
+      AdventureResult selectedItem = this.currentOrSelectedItem(slot);
+      GearChangePanel.updateEquipmentList(model, items, selectedItem);
+      this.equipment[slot].setEnabled(
+          this.isEnabled && !KoLCharacter.getLimitMode().limitSlot(slot));
+
+      if (slot == EquipmentManager.WEAPON) {
+        // Equipping 2 or more handed weapon: nothing in off-hand
+        if (EquipmentDatabase.getHands(selectedItem.getItemId()) > 1) {
+          this.equipment[EquipmentManager.OFFHAND].setSelectedItem(EquipmentRequest.UNEQUIP);
+          this.equipment[EquipmentManager.OFFHAND].setEnabled(false);
+        }
+      }
+    }
+  }
+
+  private void updateEquipmentModels(final List<List<AdventureResult>> equipmentLists) {
+    if (SwingUtilities.isEventDispatchThread() || GraphicsEnvironment.isHeadless()) {
+      updateEquipmentModelsInternal(equipmentLists);
+    } else {
+      try {
+        SwingUtilities.invokeAndWait(() -> updateEquipmentModelsInternal(equipmentLists));
+      } catch (Exception ie) {
+      }
+    }
+  }
+
+  // For performance reasons, allow deferral of updateAllModels, which recomputes the lists that
+  // show in
+  // GearChangeFrame. This way we can perform multiple equipment updates at once without triggering
+  // several runs
+  // through the entire list of equipment in a player's inventory to see if it is equippable.
+  public static void deferUpdate() {
```

Comment:
Could you fix the formatting of that comment, please?
Looks like spotlessapply broke long lines into two.

Reply:
Will do.